### PR TITLE
ocamlPackages.tls: 0.13.1 -> 0.13.2

### PR DESCRIPTION
**Pull request by superherointj**
* ocamlPackages.tls: refactoring
* ocamlPackages.tls: 0.13.1 -> 0.13.2
* ocamlPackages.tls-async: init 0.13.2

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125882` at 977b37b9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>dune-release</li>
    <li>jackline</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/tls/default.nix [hunk 1]
```diff
@@ -4,25 +4,43 @@
...
-    sha256 = "ca95fa59a82f7d38b0b495fc0cd1ff54e7728492a292895d0067c1ba9de81b7b";
+    sha256 = "sha256-IE6Fuvem8A3lZ/M8GLNYNwCG+v7BbPQ4QdYS+fKT50c=";
   };
 
+  minimumOCamlVersion = "4.08";
   useDune2 = true;
 
   doCheck = true;
-  checkInputs = [ cstruct-unix ounit2 alcotest ];
+  checkInputs = [
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/tls/default.nix:19]: Needs `randomconv` as well now, apparently.

<https://github.com/mirleft/ocaml-tls/compare/v0.13.1...v0.13.2#diff-4e4385427f3692f3a121bc8d987899d17f5a6090784b51f47b76423cfbd4a358R39>

**Review by sternenseemann — CHANGES_REQUESTED** [review]
Updating to 0.13.2 only makes sense if you add the `tls-async` package as well since that is the only notable change between 0.13.1 and 0.13.2.

**Comment by sternenseemann** [comment]
Also can you please do reformats in separate commits in the future? Formatting changes make changes much more cumbersome to review.

**Comment by superherointj** [comment]
@ofborg eval

File: pkgs/development/ocaml-modules/tls/default.nix [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+, alcotest, cstruct-unix, ounit2, randomconv
 }:
 
 buildDunePackage rec {
   pname = "tls";
-  version = "0.13.1";
+  version = "0.13.2";
 
-  src = fetchurl {
-    url = "https://github.com/mirleft/ocaml-tls/releases/download/v${version}/tls-v${version}.tbz";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/tls/default.nix:13]: This URL was the right one.

Comment (superherointj) [reply pkgs/development/ocaml-modules/tls/default.nix:13]: Done

**Comment by sternenseemann** [comment]
opam additionally lists as dependencies:

* ptime
* x509
* ppx_jane (not sure if nativeBuildInputs or propagatedBuildInputs though)

**Pull request merged**